### ISSUE TITLE
BPM LFO (v0.5.8): phase issues & blue LED (CLOCK) sometimes stays on

### ISSUE THREAD

**Issue by DomiKamu** [almostEric/FrozenWasteland#10]
Windows build v0.5.8, VCV Rack v0.5.1 (non-dev).

Hi,

1/ When I'm using external clock to BPM LFO, the output signal isn't in phase with clock, even at X1 (see attached picture).
2/ When you disconnect the cable from CLOCK connector, sometimes its blue LED stays turned on (depending the instant when you disconnect the cable).

![bpm lfo v0 5 8](https://user-images.githubusercontent.com/24851960/35790887-e93b0ef8-0a44-11e8-84bb-4494532d813e.png)

Regards.

Dominique.

**Comment by DomiKamu** [comment]
Also: they're many useless .svg files in /res (such master and ZINC). TIA.

**Comment by almostEric** [comment]
I refactored the reset input and how the clock input starts. That said, you can get the LFO out of phase if you change the divisor then back, because the divisor is applied to frequency immediately. I was considering not applying a division change until the LFO cycle finishes. Welcome your thoughts on this?

**Comment by DomiKamu** [comment]
Thanks Eric (I was not sure initially about your first name ;)

I'd like to add some infos: first for the test (screen capture) be sure I've used initialized module (without ratio change), at X1.

I've downloaded lastest 0.5.9 binaries (Windows), of course I'm understanding the problem for synch with external clock source in particular for multiply or divide. In fact, I'm designing a clock multiplier/divider modules (KlokSpid) as I've posted preview recently on Facebook (my name is Dominique Camus) - I'm C/C++ "newbie" despite I'm skilled developer (Delphi in particular, since many years - 25) so it's for me a very hard, but exciting challenge.

I understand perfectly for module (BPM Clock) doesn't have its embedded clock generator, of course, it doesn't know the frequency, BTW on input it needs to receive at least 2 pulses from source to determine the frequency (can be "fixed" or can be changed - chaotic - in real time). This is required to generate output signal at proper frequency ratio (IMO).

Also Eric be sure I consider your modules, because their concepts are original, for example Seriously Slow LFO for 100 months, simply excellent. I don't produce any music, VCV Rack is for me like a game, for experiment, and nothing else.

Sorry for mine possible English faults, because I'm French (nobody is perfect) :-) :-)

Regards,
Dominique.
